client/data/deposits/hooks.ts

```diff
@@ -10,8 +10,12 @@ import moment from 'moment';
  * Internal dependencies
  */
 import { STORE_NAME } from '../constants';
+import { Deposit } from '../../types/deposits';
+import { Query } from '@woocommerce/navigation';
 
-export const useDeposit = ( id ) =>
+export const useDeposit = (
```

Comment:
I'd avoid touching `client/data` folder at all - we didn't build enough team expertise in TS yet.

Reply:
We had the same discussion last HACK week and decided to actually touch `client/data` folder. Otherwise, we would have to define types in usage places and with time as the adoption grows we'd have the type definition in multiple places.

Reply:
Fair enough =)